docs/dev/commitlog-file-format.md

```diff
@@ -114,3 +114,33 @@ the file is not fully written/prematurely ended. Both cases can mean data loss,
 how writing is done as well as OS and hardware.
 
 ```
+
+Version 4
+---------
+
+Modified from v3 to allow fragmented data entries, i.e. writing a single data entry
+as a stream across several segments.
+A fragmented entry is written by splitting data into sub-parts that will fit into
+the normal restrictions of a write (i.e. smaller than max mutation size, but also
+trying to fit into existing buffers as best we can to avoid wasting alignment slack).
+
+```
+        Fragmented entry
+
+        magic           : fragmented marker - 0xfffffffe (MAX_UINT32-1)
+        size            : size of this fragmented entry part + headers
+        id              : the stream id
+        offset          : offset of this entry in the data stream
+        remaining       : stream data remaining to write after this entry 
+        crc             : CRC32 of magic, size, id, offset and remaining
```

Comment:
Perhaps use a hardware-accelerated version of crc32? Or this one is already in use? 
A good practice is to include the previous crc32 into the next one, this allows detecting a block which was mistakenly overwritten but is otherwise correct (e.g. due to a software bug).

Reply:
Our crc32c is hardware accelerated.